src/lib/accessibility/types.ts

```diff
@@ -140,12 +140,17 @@ export type FocusZoneDefinition = {
   props?: FocusZoneProps
 }
 
+export type FocusTrapZoneDefinition = {
+  props?: FocusTrapZoneProps
+}
+
 export type KeyActions = { [partName: string]: { [actionName: string]: KeyAction } }
 export interface AccessibilityDefinition {
   attributes?: AccessibilityAttributesBySlot
   keyActions?: KeyActions
   handledProps?: (keyof AccessibilityAttributes)[]
   focusZone?: FocusZoneDefinition
+  focusTrapZone?: FocusTrapZoneDefinition
```

Comment:
note that this becomes harder for the client to immediately get an idea of why these two are separated and what, actually, drives this division - as both are related to focus handling aspects. Maybe we should take it as a note for the future and address this (API interface) problem